sklearn/utils/testing.py

```diff
@@ -625,8 +625,9 @@ def run_test(*args, **kwargs):
             import matplotlib
             matplotlib.use('Agg', warn=False)
             # this fails if no $DISPLAY specified
-            matplotlib.pylab.figure()
-        except:
+            import matplotlib.pyplot as plt
+            plt.figure()
+        except ImportError:
```

Comment:
I guess the tests assume that matplotlib wasn't imported prior to this. This is somewhat save, as we don't / can't import matplotlib in the codebase. But not very robust.